pkg/login/social/social.go

```diff
@@ -225,3 +233,97 @@ var GetOAuthProviders = func(cfg *setting.Cfg) map[string]bool {
 
 	return result
 }
+
+func GetOAuthHttpClient(name string) (*http.Client, error) {
+	name = strings.TrimPrefix(name, "oauth_")
+
+	info, ok := setting.OAuthService.OAuthInfos[name]
+	if !ok {
+		return nil, fmt.Errorf("Could not find %s in OAuth Settings", name)
+	}
+
+	// handle call back
+	tr := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: info.TlsSkipVerify,
+		},
+	}
+	oauthClient := &http.Client{
+		Transport: tr,
+	}
+
+	if info.TlsClientCert != "" || info.TlsClientKey != "" {
+		cert, err := tls.LoadX509KeyPair(info.TlsClientCert, info.TlsClientKey)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to setup TlsClientCert error: %s", err)
+		}
+
+		tr.TLSClientConfig.Certificates = append(tr.TLSClientConfig.Certificates, cert)
+	}
+
+	if info.TlsClientCa != "" {
+		caCert, err := ioutil.ReadFile(info.TlsClientCa)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to setup TlsClientCa error: %s", err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		tr.TLSClientConfig.RootCAs = caCertPool
+	}
+	return oauthClient, nil
+}
+
+var GetCurrentOAuthToken = func(ctx context.Context, user models.SignedInUser) (*oauth2.Token, error) {
+	authInfoQuery := &models.GetAuthInfoQuery{UserId: user.UserId}
+	if err := bus.Dispatch(authInfoQuery); err != nil {
+		return nil, fmt.Errorf("Error fetching oauth information for userid:%d username:%s error:%s", user.UserId, user.Login, err)
+	}
+
+	provider := authInfoQuery.Result.AuthModule
+	connect, ok := SocialMap[strings.TrimPrefix(provider, "oauth_")] // The socialMap keys don't have "oauth_" prefix, but everywhere else in the system does
+	if !ok {
+		return nil, fmt.Errorf("Failed to find oauth provider with given name:%s", provider)
+	}
+
+	client, err := GetOAuthHttpClient(strings.TrimPrefix(provider, "oauth_"))
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create http client for oauth operation error:%s", err)
+	}
+	ctx = context.WithValue(ctx, oauth2.HTTPClient, client)
```

Comment:
I saw an http ssl error on getting a new token (refresh) and then found that login_oauth.go was setting the http client into the context.  It made sense that the same thing should be done for refreshing the token.

Reply:
Moved the OAuth HTTP Client changes to a new PR -> https://github.com/grafana/grafana/pull/27523
It will be a lot easier to include that new call on this PR once that PR is approved and merged.
I will back that code out of this PR (except for calling it once it's merged in)

Reply:
Great. I have reviewed #27523